tests/test_recommendation.py

```diff
@@ -1,14 +1,143 @@
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:expandtab
 
-from muacrypt.recommendation import Recommendation
+from muacrypt import mime
+
+
+def send_ac_mail(sender, recipient, Date=None):
+    mail = mime.gen_mail_msg(
+        From=sender.addr, To=[recipient.addr],
+        Autocrypt=sender.make_ac_header(recipient.addr),
+        Date=Date)
+    recipient.process_incoming(mail)
+
+
+def send_enc_ac_mail(sender, recipient):
+    msg = mime.gen_mail_msg(
+        From=sender.addr, To=[recipient.addr],
+        Autocrypt=sender.make_ac_header(recipient.addr))
+    r = sender.encrypt_mime(msg, [recipient.addr])
+    recipient.process_incoming(r.enc_msg)
+    r = recipient.decrypt_mime(r.enc_msg)
+    return r.dec_msg
+
+
+def send_no_ac_mail(sender, recipient):
+    mail = mime.gen_mail_msg(
+        From=sender.addr, To=[recipient.addr],
+    )
+    recipient.process_incoming(mail)
+
+
+def get_recommendation(composer, peers, reply_to_enc=False):
+    if not isinstance(peers, set):
+        peers = {peers}
+    peer_addrs = {peer.addr for peer in peers}
+    return composer.get_recommendation(peer_addrs, reply_to_enc=reply_to_enc)
 
 
 class TestRecommendation:
 
-    def test_empty_peer_state(self, account_maker):
-        sender, recipient = account_maker(), account_maker()
-        peerstate = sender.get_peerstate(recipient.addr)
-        rec = Recommendation({recipient.addr: peerstate})
-        assert rec.target_keys()[recipient.addr] is None
+    def test_disable_on_initial_mail(self, account_maker):
+        composer, peer = account_maker(), account_maker()
+        rec = get_recommendation(composer, peer)
+        assert rec.target_keys()[peer.addr] is None
+        assert rec.ui_recommendation() == 'disable'
+
+    def test_available_after_receiving_ac_mail(self, account_maker):
+        composer, peer = account_maker(), account_maker()
+        send_ac_mail(peer, composer)
+        rec = get_recommendation(composer, peer)
+        peer_key = composer.get_peerstate(peer.addr).public_keyhandle
+        assert rec.target_keys()[peer.addr] == peer_key
```

Comment:
also here i suggest to spell out "peer_keyhandle" instead of "peer_key" because "key" is ambigous -- the rest of the code bases avoids it in favor of "keyhandle" and "keydata", so that one never needs to think too much what one has.